charts/alertmanager/templates/alertmanager-networkpolicy.yaml

```diff
@@ -26,12 +26,22 @@ spec:
           tier: monitoring
           component: prometheus
           release: {{ .Release.Name }}
+    {{- if .Values.global.authSidecar.enabled  }}
+    - namespaceSelector:
+        matchLabels:
+          network.openshift.io/policy-group: ingress
+    {{- else }}
     - podSelector:
         matchLabels:
           tier: nginx
           component: ingress-controller
           release: {{ .Release.Name }}
+    {{- end }}
     ports:
     - protocol: TCP
       port: {{ .Values.ports.http }}
+    {{- if .Values.global.authSidecar.enabled  }}
+    - protocol: TCP
+      port: 8080
+    {{- end }}
```

Comment:
I feel like we should have either/or, not both ports here?

Reply:
This is to avoid internal service connection failure. if we connect to nginx-ui ( it needs an authenticated connection).